zad1/main.rb

```diff
@@ -1,13 +1,28 @@
 def sort_letters(string)
-
+    return string.split("").sort().join
 end
 
 def count_vowels(string)
-
+    vowels = 'eyuioa'
+    odpowiedz = 0
+    string.split("").each {
+        |litera|
+        if vowels.include?(litera)
+            odpowiedz = odpowiedz + 1
+        end
+    }  
+    return odpowiedz
 end
 
 def filter_odd(array)
-
+    odpowiedz = Array.new
+    array.each {
```

Comment:
Możemy się pozbyć zmiennej `odpowiedz` przez wykorzystanie metody inject z prezentacji. Inject sam nam tworzy obiekt, który zwróci na końcu i wtedy `odpowiedz` nam odpada. Można to oczywiście zrobić jeszcze ładniej, Array w Ruby posiada masę metod do właśnie takiego filtrowania np. [polecany select](https://ruby-doc.org/core-2.2.0/Array.html#method-i-select), ale Array jest też enumerablem więc ma posiada dużo metod do operacji na kolekcjach które można znaleźć [tutaj](https://ruby-doc.org/core-2.5.1/Enumerable.html)